lib/Services/GrantService.php

```diff
@@ -28,27 +28,49 @@ class GrantService implements GrantRoutes
 
     private PendingGrantTransformer $pendingTransformer;
 
+    private string $clientUrl = '';
```

Comment:
The client URL should always be defined.
```suggestion
    private string $clientUrl;
```